### ISSUE TITLE
Change main branch name

### ISSUE THREAD

**Issue by Bastian** [blacklistisnotracist/the-most-politically-correct-software#1]
The main branch of your software is called `master` branch. This might be offensive for some people.
I would suggest another name instead, for example `main` or `release`.

**Comment by blacklistisnotracist** [comment]
@Bastian 

Thank you for pointing that out! I totally missed it.

Yeah, `master` could really be offensive for some people. It could potentially reminds people of the horrendous slavery history...

I really appreciate your suggestions, but I would like to point out that `main` or `release` could also be offensive for some people. For instance, actors who have never played a main role could be offended by using `main` because it could remind them of their failures. `release` could also be offensive for some people because it kind of has some connotations where something has not been released until being on that branch, which could potentially reminds some people of our horrendous slavery history.

Hmm, I wonder if there's any name that would never offend anyone....

**Comment by Bastian** [comment]
Good point.

Maybe the branch name should be a randomly generated string then?
What about `WOBsEqXQPr`? But I'm afraid that it might insult people from countries that use a non-latin alphabet.

So, maybe instead of using an actual string, a unicode emoji would be more appropiate. I would suggest the neutral and color-less (to not offend color-blinds) white square emoji: ⬜️

**Comment by Bastian** [comment]
A friend just told me, that a **white** can indeed be offensive as well.
So instead, maybe 🌈 ?

**Comment by blacklistisnotracist** [comment]
Hmm, a random string could offend some people because it's often not readable.

> But I'm afraid that it might insult people from countries that use a non-latin alphabet.

Good point!

But the term emoji came from Japanese, which could potentially hurt Americans or Asians who got hurt in WW2, so may not be the perfect characters to use... Hmm....

> A friend just told me, that a white square can indeed be offensive as well.
So instead, maybe 🌈 ?

Agreed. No mentioning of colors is a must. Otherwise some people will get furious to remove all colors...
Rainbow emoji looks cute! LBGTQ+ communities also like to use rainbow so it definitely is a good symbol of diversity and inclusion.

.... But wait, what about those who have never seen rainbows? Setting a rainbow as the default branch could hurt someone's feelings because that could imply that having seen a rainbow is the norm (default).

**Comment by blacklistisnotracist** [comment]
Wow, this is a thing! GitHub's CLI changed the default branch from master to trunk. I still think `trunk` is not enough....

https://github.com/cli/cli/issues/929

**Comment by Bastian** [comment]
What about 💩?
Every human being should produce 💩 and thus it unites us all?

**Comment by tzyganu** [comment]
What if :rainbow: offends color blind people?
You're bragging about seeing all kinds of colors when other cannot. Not cool "person", not cool.

**Comment by blacklistisnotracist** [comment]
@Bastian 

> What about 💩?
Every human being should produce 💩 and thus it unites us all?

💩  could be offensive for ostomates or people with digesting disabilities because 💩  implicitly assumes your bowels to be healthy....


@tzyganu 

> What if 🌈 offends color blind people?
You're bragging about seeing all kinds of colors when other cannot. Not cool "person", not cool.

Good point, I totally missed it! What do you think would be a good fit to our default branch?

**Comment by tzyganu** [comment]
first of all stop calling it "default". It implies that when something is not explicitly stated, there is an option that has priority / privilege over others. So "default" has to go. 
Sorry, I don't have anything in mind, but I bet I can find something offensive to someone / some thing in every name you pick. (which is the trend now, right?).

**Comment by blacklistisnotracist** [comment]
> first of all stop calling it "default". It implies that when something is not explicitly stated, there is an option that has priority / privilege over others. So "default" has to go.

In that case we need to rewrite GitHub... Could we make a PR for that?

> Sorry, I don't have anything in mind, but I bet I can find something offensive to someone / some thing in every name you pick. (which is the trend now, right?).

Exactly. Git is such a racist software...

**Comment by tzyganu** [comment]
> In that case we need to rewrite GitHub... Could we make a PR for that?

Not without offending someone

**Comment by Gallardo994** [comment]
What if we use dark-grey box? It's neither white, nor black, nor colorful to offend blind people, and it's not gender-specific.

There's also an option of using "⠀", as a completely empty character which will not offend anyone. Git also supports that.

**Comment by blacklistisnotracist** [comment]
@Gallardo994 Thank you for your suggestions!

> What if we use dark-grey box? It's neither white, nor black, nor colorful to offend colorblind people, and it's not gender-specific.

Nice idea, but it could offend gray people: https://www.urbandictionary.com/define.php?term=gray%20people

> A gray person is someone who is very uneducated,low class,whose vocabulary consists of the the F word, and who dropped out of high school with one goal in mind (i.e. to work at an assembly line for the automotive industry). For those who don't know, Southeast Michigan is full of gray people. People who thought it was their birthright to have a factory job. Gray people in general are very ignorant people who are over aggressive and want to pick fights with people that they perceive to be different. These people have very narrow minds and typically cannot think outside the box. They dress similiar and also act the same way. Grey people are very much groupies and very uneducated.

> There's also an option of using "⠀", as a completely empty character which will not offend anyone. Git also supports that.

Interesting. I wonder if it could be interpreted as the final product being empty (or useless), though. Also, whitespace is usually interchangeably used to refer an empty space, so using an empty char might be potentially attributed to white supremacy...

**Comment by Bastian** [comment]
A **white**space? Are you guys crazy?

**Comment by tzyganu** [comment]
@blacklistisnotracist I think this issue can be closed now. It was escalated to a "higher power": https://www.zdnet.com/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/

**Comment by blacklistisnotracist** [comment]
@tzyganu Doesn't that mean we should put more efforts in finding the one???

**Comment by blacklistisnotracist** [comment]
What if we name the branch `censored` or `*****`?

**Comment by stinovlas** [comment]
I think we should ask git to allow empty branch names, because empty string is perfectly politically correct.

**Comment by stinovlas** [comment]
Also, you should just rename the project to empty string as well, to avoid offending anyone by the words "politically" or "correct". People who had to spend time in correctional facilities, to name a few.

**Comment by jrial** [comment]
> What if we use dark-grey box? It's neither white, nor black, nor colorful to offend colorblind people, and it's not gender-specific.

Nope. Unfortunately, that would be offensive to the Dutch and Flemish, as in our language, "box" is a derogatory term used for either a female's genitalia, or an old and sexually unattractive woman when preceded by the adjective "old". Also, "grey" is offensive to the elderly who haven't come to grips with the passage of time yet.

But I think we're going about this all wrong. Discrimination is not the act of offending _someone_. It's the act of treating some people differently from the rest. Since we already established that it's impossible to please everyone, how about we come up with a branch name that offends everyone in equal measure, regardless of race, colour,  sexual identity, sexual preference, religion, socioeconomic status, ability, intelligence, political affiliation etc...?

It will be super offensive, but at least everyone will be equally offended and have no reason to complain about being discriminated against.

**Comment by MattIBM** [comment]
we can't have primary because of **primary colours**. How about default? Surely nothing can be linked to the word **default**... wait it's in English... it could be offensive to anyone who cannot speak english. Guess the only option we have left is to delete all our repositories.

**Comment by blacklistisnotracist** [comment]
@jrial

> Since we already established that it's impossible to please everyone, how about we come up with a branch name that offends everyone in equal measure, regardless of race, colour, sexual identity, sexual preference, religion, socioeconomic status, ability, intelligence, political affiliation etc...?

That's a good point. What about "git"? git is banned by The House of Commons of the United Kingdom.

> Words to which objection has been taken by the Speaker over the years include blackguard, coward, git, guttersnipe, hooligan, rat, swine, stoolpigeon and traitor.

https://www.parliament.uk/site-information/glossary/unparliamentary-language/

**Comment by jrial** [comment]
Won't offend Linux Torvalds; in fact, he might take it as a compliment: https://git.wiki.kernel.org/index.php/Git_FAQ#Why_the_.27Git.27_name.3F

**Comment by Gallardo994** [comment]
Does this mean we should fork mercurial?

**Comment by blacklistisnotracist** [comment]
> Does this mean we should fork mercurial?

I'm not 100% sure if he's going to get offended by this.

> Linus Torvalds, in his Google Tech Talk on revision control systems, pointed out Mercurial as the only other tool than git (the system he designed himself) worthy of consideration. 

https://www.mercurial-scm.org/wiki/BzrVsHg

What about doing like `you-are-dumb-or-bug`? [Linus would definitely get pissed off when there's a bug](https://lkml.org/lkml/2012/12/23/75). Also, by extending the condition with an OR, we can make sure to include those who would not get offended by a bug.

**Comment by Zoldszemesostoros** [comment]
What about a zero-length Unicode character?

**Comment by aayushjain** [comment]
How about using `null` for the default branch name?

**Comment by LuvForAirplanes** [comment]
> How about using `null` for the default branch name?

This way, we would all be equally offended, and no one has to feel left out.

**Comment by daaniiieel** [comment]
Null implies something that doesn't have a value or doesn't exist -- so it would offend people that don't exist and people that don't exist **yet** (people that will be born after this). Very offensive -- wouldn't recommend.

**Comment by jrial** [comment]
> ... and people that don't exist **yet** (people that will be born after this).

Hah, someone read James Morrow's "This is the Way the World Ends", have we? 🙂

**Comment by aayushjain** [comment]
We can rename `master` to `higgs-boson`, which is one of the elementary particles that give mass to matter — just like the jobless people, belabouring a trivial issue with their insipid arguments. 

Instead of addressing the root cause of an ancient and still-prevalent societal-ill, we have people working up the most meaningless and easiest solution to feel better about themselves.

**Comment by tzyganu** [comment]
@LuvForAirplanes 

> This way, we would all be equally offended, and no one has to feel left out.

I am not offended by the word `null`. So the fact that you choose a name that offends everyone else except me makes me feel left out. So I am double offended by that, hence `null` won't work.  

@aayushjain 

`higgs-boson` won't work either. Some people refer to this as `god particle`. So it will be offensive for the people that follow about 4000 religions.  

I have another idea.  
Let's delete all our repos and quit coding for good.  If we cannot find a simple branch name that will not offend anyone it means coding is totally unacceptable.

**Comment by JamesDavid1997** [comment]
Let us just eradicate Java from existence

**Comment by crossing** [comment]
Sorry for being off the topic, but I noticed the word '**default**' is used a lot in the discussion. I'm afraid it might recall traumatizing memories of people who have or had debt problems.

**Comment by Aspie96** [comment]
I suggest calling it "feminism"?
Feminism sure offends somebody, but you have to be a white cis het man (a mysoginist) to be offended by equality.
Please rename the "master" branch into "feminism".

**Comment by jrial** [comment]
> I suggest calling it "feminism"?
> Feminism sure offends somebody, but you have to be a white cis het man (a mysoginist) to be offended by equality.
> Please rename the "master" branch into "feminism".

No, no, we were doing lighthearted ridicule, not thinly veiled bitter rant... Get with the program, man!

**Comment by Gallardo994** [comment]
We are here for making serious decisions, don't forget about that please.
What candidates do we have right now afterall?

**Comment by BlNotRacist** [comment]
FYI blacklistisnotracist has been suspended.

@Gallardo994 

> What candidates do we have right now afterall?

I don't think we have come to any conclusion yet. At first we tried to find a word that does not offend anyone but it turned out it was impossible. Now we're trying to find a word that would *equally* offend anyone, but having trouble there as well. This world is full of challenging problems, and this issue is one of the hardest ones...

**Comment by BlNotRacist** [comment]
I have created a new organization and moved this repo to there: https://github.com/Blacklist-is-not-a-racist-word/the-most-politically-correct-software

**Comment by nkakouros** [comment]
What about the silence of the last 16 days? That surely offends noone. How do you "code" this into a branch name?
